WooCommerce/Classes/Authentication/Store Creation/StoreCreationStoreSwitchScheduler.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+import Yosemite
+import Combine
+
+protocol StoreCreationStoreSwitchScheduler {
+    func savePendingStoreSwitch(siteID: Int64, expectedStoreName: String)
+
+    func removePendingStoreSwitch()
+
+    var isPendingStoreSwitch: Bool { get }
+
+    func listenToPendingStoreAndReturnSiteIDOnceReady() async throws -> Int64?
+}
+
+/// Stores store creation info and informs once the store is ready
+///
+final class DefaultStoreCreationStoreSwitchScheduler: StoreCreationStoreSwitchScheduler {
+    private let stores: StoresManager
+    private let userDefaults: UserDefaults
+    private var storeStatusChecker: StoreCreationStatusChecker?
+    private var jetpackSiteSubscription: AnyCancellable?
+
+    private var siteIDPendingStoreSwitch: Int64? {
+        set {
+            userDefaults[.siteIDPendingStoreSwitch] = newValue
+        }
+        get {
+            userDefaults[.siteIDPendingStoreSwitch] as? Int64
+        }
+    }
+
+    private var expectedStoreNamePendingStoreSwitch: String? {
+        set {
+            userDefaults[.expectedStoreNamePendingStoreSwitch] = newValue
+        }
+        get {
+            userDefaults[.expectedStoreNamePendingStoreSwitch] as? String
+        }
+    }
+
+    init(stores: StoresManager = ServiceLocator.stores,
+         userDefaults: UserDefaults = .standard,
+         storeStatusChecker: StoreCreationStatusChecker? = nil) {
+        self.stores = stores
+        self.userDefaults = userDefaults
+        self.storeStatusChecker = storeStatusChecker
+    }
+
+    func savePendingStoreSwitch(siteID: Int64, expectedStoreName: String) {
+        siteIDPendingStoreSwitch = siteID
+        expectedStoreNamePendingStoreSwitch = expectedStoreName
+    }
+
+    func removePendingStoreSwitch() {
+        siteIDPendingStoreSwitch = nil
+        expectedStoreNamePendingStoreSwitch = nil
+    }
+
+    var isPendingStoreSwitch: Bool {
+        guard siteIDPendingStoreSwitch != stores.sessionManager.defaultStoreID else {
+            siteIDPendingStoreSwitch = nil
+            expectedStoreNamePendingStoreSwitch = nil
```

Comment:
Nit: should we replace this with `removePendingStoreSwitch()`?

Reply:
Good thinking. Done in e9d554a88347be4614cc5f4f35946ab2879fb1e1